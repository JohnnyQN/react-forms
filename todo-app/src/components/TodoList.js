import React, { useState } from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";

function TodoList() {
  const [todos, setTodos] = useState([]);

  // Function to add a new todo
  const addTodo = (newTodo) => {
    setTodos([...todos, newTodo]);
  };

  // Function to remove a todo by its ID
  const removeTodo = (id) => {
    setTodos(todos.filter(todo => todo.id !== id));
  };

  return (
    <div>
      <h2>Your Todos</h2>
      <NewTodoForm createTodo={addTodo} />
      <ul>
        {todos.map(todo => (
          <Todo key={todo.id} id={todo.id} task={todo.task} removeTodo={removeTodo} />
        ))}
      </ul>
    </div>
  );
}

export default TodoList;
