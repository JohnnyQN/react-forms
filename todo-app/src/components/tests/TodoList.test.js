import { render, fireEvent } from "@testing-library/react";
import TodoList from "../TodoList";

// Smoke test
test("renders without crashing", () => {
  render(<TodoList />);
});

// Snapshot test
test("matches snapshot", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

// Test adding and removing todos
test("can add and remove todos", () => {
  const { getByLabelText, getByText, queryByText } = render(<TodoList />);

  // Add a todo
  const input = getByLabelText("Enter a new task:");
  fireEvent.change(input, { target: { value: "New Todo" } });
  fireEvent.click(getByText("Add Todo"));

  // Check that the todo is added
  expect(getByText("New Todo")).toBeInTheDocument();

  // Remove the todo
  fireEvent.click(getByText("X"));

  // Check that the todo is removed
  expect(queryByText("New Todo")).not.toBeInTheDocument();
});
