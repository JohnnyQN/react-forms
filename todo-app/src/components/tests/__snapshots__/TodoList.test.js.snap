// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Your Todos
    </h2>
    <form>
      <label
        for="task"
      >
        Enter a new task:
      </label>
      <input
        id="task"
        name="task"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
